app-modules/service-management/src/Filament/Resources/ContractTypeResource/Pages/CreateContractType.php

```diff
@@ -0,0 +1,96 @@
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\ServiceManagement\Filament\Resources\ContractTypeResource\Pages;
+
+use Filament\Forms\Form;
+use Illuminate\Support\Facades\DB;
+use Filament\Forms\Components\Toggle;
+use Filament\Forms\Components\TextInput;
+use Filament\Resources\Pages\CreateRecord;
+use AidingApp\ServiceManagement\Models\ContractType;
+use AidingApp\ServiceManagement\Filament\Resources\ContractTypeResource;
+
+class CreateContractType extends CreateRecord
+{
+    protected static string $resource = ContractTypeResource::class;
+
+    public function form(Form $form): Form
+    {
+        return $form
+            ->schema([
+                TextInput::make('name')
+                    ->required(),
+                Toggle::make('is_default')
+                    ->label('Default')
+                    ->live()
+                    ->hint(function (?ContractType $record, $state): ?string {
+                        if ($record?->is_default) {
+                            return null;
+                        }
+
+                        if (! $state) {
+                            return null;
+                        }
+
+                        $currentDefault = ContractType::query()
+                            ->where('is_default', true)
+                            ->value('name');
+
+                        if (blank($currentDefault)) {
+                            return null;
+                        }
+
+                        return "The current default status is '{$currentDefault}', you are replacing it.";
+                    })
+                    ->hintColor('danger')
+                    ->columnStart(1),
+            ]);
+    }
+
+    protected function mutateFormDataBeforeCreate(array $data): array
+    {
+        $data['order'] = $data['order'] = DB::table('contract_types')->max('order') + 1;
+
+        if ($data['is_default']) {
+            ContractType::query()
+                ->where('is_default', true)
+                ->update(['is_default' => false]);
+        }
```

Comment:
Let's add `protected ?bool $hasDatabaseTransactions = true;` to this page so that if the Contract type you are creating fails to save for some reason this change is rolled back.